django/contrib/auth/backends.py

```diff
@@ -119,6 +120,31 @@ def has_module_perms(self, user_obj, app_label):
             for perm in self.get_all_permissions(user_obj)
         )
 
+    def with_perm(self, perm, is_active=True, is_superuser=True, obj=None):
+        if not isinstance(perm, (str, Permission)):
+            raise TypeError('The `perm` argument must be a string or a permission instance.')
+        elif isinstance(perm, str):
+            try:
+                app_label, codename = perm.split('.')
+            except ValueError:
+                raise ValueError("Permission name should be in the form 'app_label.perm_name'.") from None
+
+        UserModel = get_user_model()
+
+        if obj is not None:
+            return UserModel._default_manager.none()
+
+        if isinstance(perm, Permission):
+            q = Q(pk=perm.pk)
+        else:
+            q = Q(codename=codename, content_type__app_label=app_label)
+        q &= Q(group__user=OuterRef('pk')) | Q(user=OuterRef('pk'))
+        return UserModel._default_manager.annotate(
+            has_permission=Exists(Permission.objects.filter(q)),
+        ).filter(
+            Q(is_active=is_active) & (Q(is_superuser=is_superuser) | Q(has_permission=True)),
```

Comment:
The `Q(is_superuser=...)` should only be added here if `include_superuser=True`. Otherwise we get all non-superuser users returned when we pass `False`.